discovery-provider/plugins/notifications/src/processNotifications/mappers/announcement.ts

```diff
@@ -13,84 +13,112 @@ type AnnouncementNotificationRow = Omit<NotificationRow, 'data'> & {
   data: AnnouncementNotification
 }
 export class Announcement extends BaseNotification<AnnouncementNotificationRow> {
-  receiverUserIds: number[]
 
   constructor(
     dnDB: Knex,
     identityDB: Knex,
     notification: AnnouncementNotificationRow
   ) {
     super(dnDB, identityDB, notification)
-    this.receiverUserIds = this.notification.user_ids!
   }
 
   async pushNotification() {
-    const res: Array<{
-      user_id: number
-      name: string
-      is_deactivated: boolean
-    }> = await this.dnDB
-      .select('user_id', 'name', 'is_deactivated')
-      .from<UserRow>('users')
+    const pushAnnouncements = false;
+    if (!pushAnnouncements) {
+      // TODO: ADD .env CONFIG TO SHUT OFF ANNOUNCEMENTS
+      return;
+    }
+
+    const res_count = await this.dnDB('users')
+      .count('user_id')
       .where('is_current', true)
-      .whereIn('user_id', this.receiverUserIds)
+      .where('is_deactivated', false)
+      .first()
 
-    const users = res.reduce((acc, user) => {
-      acc[user.user_id] = {
-        name: user.name,
-        isDeactivated: user.is_deactivated
-      }
-      return acc
-    }, {} as Record<number, { name: string; isDeactivated: boolean }>)
+    // this isn't good if the res is a string
+    const count = res_count.count as number
+    let offset = 0 // let binding because we re-assign
+    const page_count = 10000
 
-    // TODO: Batch send out notifications
+    const total_start = new Date().getTime()
 
-    const validReceiverUserIds = this.receiverUserIds.filter(
-      (userId) => !(users?.[userId]?.isDeactivated ?? true)
-    )
-    for (const userId of validReceiverUserIds) {
-      const userNotificationSettings = await buildUserNotificationSettings(
-        this.identityDB,
-        [userId]
+    // this will leave out a few extra people
```

Comment:
also is there any way to add in those few extra people at the end? ideally we really get all

Reply:
oh! meant to fix that, whoops